pkg/sync/syncer.go

```diff
@@ -282,16 +282,23 @@ func (s *syncer) startOrResumeSync(ctx context.Context) (string, bool, error) {
 	var newSync bool
 	var err error
 	if len(s.targetedSyncResourceIDs) == 0 {
-		syncID, newSync, err = s.store.StartSync(ctx)
-		if err != nil {
-			return "", false, err
+		if s.skipEntitlementsAndGrants {
```

Comment:
maybe just set the synctype in the condition, and do the rest outside of the branching logic (s.store.StartOrResumeSync...)